pypeit/spectrographs/keck_hires.py

```diff
@@ -28,22 +49,107 @@ class KECKHIRESSpectrograph(spectrograph.Spectrograph):
+    header_name = 'HIRES'
     pypeline = 'Echelle'
+    ech_fixed_format = False
+    supported = True
 
+
+    # Place holder taken from X-shooter VIS
     @classmethod
     def default_pypeit_par(cls):
         """
         Return the default parameters to use for this instrument.
-        
+
         Returns:
             :class:`~pypeit.par.pypeitpar.PypeItPar`: Parameters required by
             all of ``PypeIt`` methods.
         """
         par = super().default_pypeit_par()
-        # Correct for flexure using the default approach
-        par['flexure'] = pypeitpar.FlexurePar()
+
+        par['rdx']['detnum'] = [(1,2,3)]
+
+        # Adjustments to parameters for VIS
+        turn_on = dict(use_biasimage=False, use_overscan=True, overscan_method='median',
+                       use_darkimage=False, use_illumflat=False, use_pixelflat=False,
+                       use_specillum=False)
+        par.reset_all_processimages_par(**turn_on)
+        # X-SHOOTER arcs/tilts are also have different binning with bias
+        # frames, so don't use bias frames. Don't use the biases for any
+        # calibrations since it appears to be a different amplifier readout
+        par['calibrations']['traceframe']['process']['overscan_method'] = 'median'
+
+        # Right now we are using the overscan and not biases becuase the
+        # standards are read with a different read mode and we don't yet have
+        # the option to use different sets of biases for different standards,
+        # or use the overscan for standards but not for science frames
+        # TODO testing
+        par['scienceframe']['process']['use_biasimage'] = False
+        par['scienceframe']['process']['use_illumflat'] = False
+        par['scienceframe']['process']['use_pixelflat'] = False
+        par['calibrations']['standardframe']['process']['use_illumflat'] = False
+        par['calibrations']['standardframe']['process']['use_pixelflat'] = False
+        # par['scienceframe']['useframe'] ='overscan'
+
+        par['calibrations']['slitedges']['edge_thresh'] = 8.0
+        par['calibrations']['slitedges']['fit_order'] = 8
+        par['calibrations']['slitedges']['max_shift_adj'] = 0.5
+        par['calibrations']['slitedges']['trace_thresh'] = 10.
+        par['calibrations']['slitedges']['left_right_pca'] = True
+        par['calibrations']['slitedges']['length_range'] = 0.3
+        par['calibrations']['slitedges']['max_nudge'] = 0.
+        par['calibrations']['slitedges']['overlap'] = True
+        par['calibrations']['slitedges']['dlength_range'] = 0.25
+
+        # These are the defaults
+        par['calibrations']['tilts']['tracethresh'] = 15
+        par['calibrations']['tilts']['spat_order'] = 3
+        par['calibrations']['tilts']['spec_order'] = 5  # [5, 5, 5] + 12*[7] # + [5]
+
+        # 1D wavelength solution
+        par['calibrations']['wavelengths']['lamps'] = ['ThAr']
+        # This is for 1x1 binning. TODO GET BINNING SORTED OUT!!
```

Comment:
I suggest we put in a hook here that errors out if the binning is not 1 in the spectral dimension.  This is otherwise going to bite us when we least expect it.

Reply:
Not sure I can figure out where that should go now.